jerry-core/ecma/operations/ecma-objects-arguments.c

```diff
@@ -140,6 +140,22 @@ ecma_op_create_arguments_object (ecma_object_t *func_obj_p, /**< callee function
 
   ecma_property_descriptor_t prop_desc = ecma_make_empty_property_descriptor ();
 
+#if ENABLED (JERRY_ES2015)
+  /* ECMAScript v6, 9.4.4.6.7, 9.4.4.7.22 */
+  ecma_string_t *symbol_p = ecma_op_get_global_symbol (LIT_MAGIC_STRING_ITERATOR);
+
+  prop_value_p = ecma_create_named_data_property (obj_p,
+                                                  symbol_p,
+                                                  ECMA_PROPERTY_CONFIGURABLE_WRITABLE,
+                                                  NULL);
+
+  prop_value_p->value = ecma_op_object_get_by_magic_id (ecma_builtin_get (ECMA_BUILTIN_ID_ARRAY_PROTOTYPE),
+                                                        LIT_MAGIC_STRING_VALUES);
```

Comment:
Is this always successfull?

Reply:
Ohh, good catch. According to the standard the arguments object creation operation never throws exception, however, this [[Get]] can, since the Array.prototype.values is function object is configurable and writable. However the standard specifies, that `Perform DefinePropertyOrThrow(obj, @@iterator, PropertyDescriptor {[[Value]]:%ArrayProto_values%, [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true}).`. Therefore the `%ArrayProto_values%` function object need to be saved after the lazy instantiation or it should be a new builtin routine.